### DISCUSSION TITLE
--open equivalent in JS API

### DISCUSSION THREAD

**Discussion by guoyunhe** [parcel-bundler/parcel#8898] [Q&A]
I have the following JS code to start dev server:

```js
let bundler = new Parcel({
  entries: '.parcel-docs/index.html',
  defaultConfig: '@parcel/config-default',
  defaultTargetOptions: {
    distDir: '.parcel-docs/dist',
  },
  serveOptions: {
    port: 3000,
  },
  hmrOptions: {
    port: 3000,
  },
});
await bundler.watch();
```

However, I didn't find a way to automatically open browser. Just like `--open` option in CLI mode. Does Parcel support this in JS API? Or should I use `open` package?

And how could I keep the same console output in Parcel API like Parcel CLI?

![图片](https://user-images.githubusercontent.com/5836790/226081195-f31fb5ca-f76e-4559-a1a5-7e5efa6701f4.png)

**Comment by guoyunhe** [comment]
My current solution is:
```js
  let bundler = new Parcel({
    entries: '.parcel-docs/index.html',
    defaultConfig: '@parcel/config-default',
    defaultTargetOptions: {
      distDir: '.parcel-docs/dist',
    },
    serveOptions: {
      port: 3000,
    },
    hmrOptions: {
      port: 3000,
    },
    additionalReporters: [
      {
        packageName: '@parcel/reporter-cli',
        resolveFrom: fileURLToPath(import.meta.url)
      },
    ],
  });
  await bundler.watch();
  await open('http://localhost:3000/');
```

Not sure if it is the best way to do it.

**Reply by mischnic** [reply]
Here's what the CLI does, so this is indeed not implemented in the core API: https://github.com/parcel-bundler/parcel/blob/b2331a19365071cd1a8103408c21523295f24013/packages/core/parcel/src/cli.js#L317-L330
